src/lib/Names.js

```diff
@@ -0,0 +1,211 @@
+const CLEAN_REGEX = /\/\//g;
+
+// http://wiki.ros.org/Names
+class Names {
+  constructor() {
+    this._remappings = {};
+    this._namespace = '';
+  }
+
+  init(remappings, namespace) {
+    this._namespace = namespace;
+    this._remappings = {};
+
+    Object.keys(remappings).forEach((left) => {
+      if (left && !left.startsWith('_')) {
+        const right = remappings[left];
+
+        const resolvedLeft = this.resolve(left, false);
+        const resolvedRight = this.resolve(right, false);
+
+        this._remappings[resolvedLeft] = resolvedRight;
+      }
+    });
+  }
+
+  validate(name, throwError = false) {
+    if (typeof name !== 'string') {
+      if (throwError) {
+        throw new Error('Unable to validate non-string name');
+      }
+      return false;
+    }
+
+    const len = name.length;
+    if (len === 0) {
+      return true;
+    }
+    // else
+    // First character must be alphanumeric, '/', or '~'
+    const c = name[0];
+    if (!isAlpha(c) && c !== '/' && c !== '~') {
+      if (throwError) {
+        throw new Error(`Character [${c}] is not valid as the first character in Graph Resource Name [${name}].  Valid characters are a-z, A-Z, / and in some cases ~.`);
+      }
+      // else
+      return false;
+    }
+
+    for (let i = 1; i < len; ++i) {
```

Comment:
RegExp test might be better for this in my opinion: 

![image](https://user-images.githubusercontent.com/6723574/42259194-47b63716-7f14-11e8-9d74-ba8e508035f5.png)

Just replace the first character with `-` maybe then the regexp should work.

DYI: https://jsperf.com/alphanumeric-string-test